// @xstyled/styled-components/dist/xstyled-styled-components.cjs.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var scStyled = require('styled-components');
var scStyled__default = _interopDefault(scStyled);
var util = require('@xstyled/util');
var core = require('@xstyled/core');
var _taggedTemplateLiteralLoose = _interopDefault(require('@babel/runtime/helpers/taggedTemplateLiteralLoose'));
var React = _interopDefault(require('react'));
var system = require('@xstyled/system');

/* eslint-disable no-continue, no-loop-func, no-cond-assign */
function css() {
  var scCssArgs = scStyled.css.apply(void 0, arguments);
  var flattenedArgs = util.flattenStrings(scCssArgs);
  return flattenedArgs.map(core.transform);
}

var createGlobalStyle = function createGlobalStyle() {
  return scStyled.createGlobalStyle([css.apply(void 0, arguments)]);
};

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["", ""]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function getCreateStyle(baseCreateStyle) {
  var createStyle = function createStyle() {
    return baseCreateStyle(_templateObject(), css.apply(void 0, arguments));
  };

  createStyle.attrs = function (attrs) {
    var nextCreateStyle = baseCreateStyle.attrs(attrs);
    return getCreateStyle(nextCreateStyle);
  };

  createStyle.withConfig = function (config) {
    var nextCreateStyle = baseCreateStyle.withConfig(config);
    return getCreateStyle(nextCreateStyle);
  };

  return createStyle;
}

function styled(component) {
  return getCreateStyle(scStyled__default(component));
}
var InnerBox = /*#__PURE__*/system.createSystemComponent(React);
var Box = /*#__PURE__*/styled(InnerBox)(core.createBox);
styled.box = /*#__PURE__*/styled(Box);
Object.keys(scStyled__default).forEach(function (key) {
  styled[key] = styled(key);
  styled[key + "Box"] = styled(Box.withComponent(system.createSystemComponent(React, key)));
});

function _templateObject$1() {
  var data = _taggedTemplateLiteralLoose(["", ""]);

  _templateObject$1 = function _templateObject() {
    return data;
  };

  return data;
}
var ColorModeStyle = /*#__PURE__*/scStyled.createGlobalStyle( /*#__PURE__*/_templateObject$1(), function (p) {
  return core.createColorStyles(p.theme, {
    targetSelector: p.targetSelector
  });
});
var ColorModeProvider = /*#__PURE__*/core.createColorModeProvider({
  ThemeContext: scStyled.ThemeContext,
  ThemeProvider: scStyled.ThemeProvider,
  ColorModeStyle: ColorModeStyle
});

function useTheme() {
  return React.useContext(scStyled.ThemeContext);
}

function useBreakpoints() {
  return core.useThemeBreakpoints(useTheme());
}
function useBreakpoint() {
  return core.useThemeBreakpoint(useTheme());
}
function useUp(key) {
  return core.useThemeUp(useTheme(), key);
}
function useDown(key) {
  return core.useThemeDown(useTheme(), key);
}

Object.keys(system).forEach(function (k) {
  if (k !== 'default') Object.defineProperty(exports, k, {
    enumerable: true,
    get: function () {
      return system[k];
    }
  });
});
Object.defineProperty(exports, 'ServerStyleSheet', {
  enumerable: true,
  get: function () {
    return scStyled.ServerStyleSheet;
  }
});
Object.defineProperty(exports, 'StyleSheetConsumer', {
  enumerable: true,
  get: function () {
    return scStyled.StyleSheetConsumer;
  }
});
Object.defineProperty(exports, 'StyleSheetContext', {
  enumerable: true,
  get: function () {
    return scStyled.StyleSheetContext;
  }
});
Object.defineProperty(exports, 'StyleSheetManager', {
  enumerable: true,
  get: function () {
    return scStyled.StyleSheetManager;
  }
});
Object.defineProperty(exports, 'ThemeConsumer', {
  enumerable: true,
  get: function () {
    return scStyled.ThemeConsumer;
  }
});
Object.defineProperty(exports, 'ThemeContext', {
  enumerable: true,
  get: function () {
    return scStyled.ThemeContext;
  }
});
Object.defineProperty(exports, 'ThemeProvider', {
  enumerable: true,
  get: function () {
    return scStyled.ThemeProvider;
  }
});
Object.defineProperty(exports, 'isStyledComponent', {
  enumerable: true,
  get: function () {
    return scStyled.isStyledComponent;
  }
});
Object.defineProperty(exports, 'keyframes', {
  enumerable: true,
  get: function () {
    return scStyled.keyframes;
  }
});
Object.defineProperty(exports, 'withTheme', {
  enumerable: true,
  get: function () {
    return scStyled.withTheme;
  }
});
Object.defineProperty(exports, 'getColorModeInitScriptElement', {
  enumerable: true,
  get: function () {
    return core.getColorModeInitScriptElement;
  }
});
Object.defineProperty(exports, 'getColorModeInitScriptTag', {
  enumerable: true,
  get: function () {
    return core.getColorModeInitScriptTag;
  }
});
Object.defineProperty(exports, 'useColorMode', {
  enumerable: true,
  get: function () {
    return core.useColorMode;
  }
});
Object.defineProperty(exports, 'useViewportWidth', {
  enumerable: true,
  get: function () {
    return core.useViewportWidth;
  }
});
exports.Box = Box;
exports.ColorModeProvider = ColorModeProvider;
exports.createGlobalStyle = createGlobalStyle;
exports.css = css;
exports.default = styled;
exports.useBreakpoint = useBreakpoint;
exports.useBreakpoints = useBreakpoints;
exports.useDown = useDown;
exports.useTheme = useTheme;
exports.useUp = useUp;
