// @react95/core/Video/buttons/Play.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

var Play = function Play(props) {
  return /*#__PURE__*/_react["default"].createElement("svg", _extends({
    height: "6",
    viewBox: "0 0 494.942 494.942",
    width: "6",
    xmlns: "http://www.w3.org/2000/svg",
    "aria-label": "play"
  }, props), /*#__PURE__*/_react["default"].createElement("path", {
    d: "M35.353 0l424.236 247.471L35.353 494.942z"
  }));
};

var _default = Play;
exports["default"] = _default;
